src/getAnimalMap.js

```diff
@@ -1,7 +1,36 @@
-const data = require('../data/zoo_data');
+const { species } = require('../data/zoo_data');
 
-function getAnimalMap(options) {
-  // seu código aqui
+function searchDefault() {
+  const regions = {};
+
+  species.forEach((specie) => {
+    if (!regions[specie.location]) regions[specie.location] = [];
+    regions[specie.location].push(specie.name);
+  });
+
+  return regions;
+}
+
+function animalNames(sorted, sex) {
+  const regions = {};
+
+  species.forEach((specie) => {
+    if (!regions[specie.location]) regions[specie.location] = [];
+    let residents = [...specie.residents];
+    if (sex) residents = residents.filter((resident) => resident.sex === sex);
+    residents = residents.map((resident) => resident.name);
+    if (sorted) residents.sort();
+    regions[specie.location].push({ [specie.name]: residents });
+  });
+
+  return regions;
+}
+
+function getAnimalMap(options = {}) {
+  const { includeNames, sorted, sex } = options;
+
+  if (includeNames) return animalNames(sorted, sex);
```

Comment:
```suggestion
  if (includeNames) return getAnimalNames(sorted, sex);
```

Mantendo o padrão de usar verbos nas função